### PR TITLE
Detect extra cases in struct declarations

### LINKED ISSUE #63: Incomplete detection for anonymous structs in function signature
We have a code base where the following code appears :

```go
func SomeHandler(args *struct{
    Arg1 int    `json:"arg1,omitempty"`
    Arg2 string `json:"arg2,omitempty"`
}) error {
    ...
```

When running `gomodifytags` with an offset falling within the range of this struct declaration, it returns an error : `offset is not inside a struct`.

### PR DESCRIPTION
**Pull request by LeGEC**
Handle the following extra cases :
* anonymous structs in function signature
* "pointer to struct" type declaration
* "array of struct" type declaration

and nested variants : "array of pointers to array of structs" ...

Fixes #63

### CODE REVIEW

**Comment by LeGEC** [comment]
first commit adds extra failing cases to highlight some missing elements,  
second commit contains the fix

File: main.go [hunk 1]
```diff
@@ -448,6 +448,25 @@ func collectStructs(node ast.Node) map[token.Pos]*structType {
...
+	loop:
+		for true {
+			switch x := t.(type) {
+			case *ast.StarExpr:
+				t = x.X
+			case *ast.ArrayType:
+				t = x.Elt
+			default:
+				break loop
+			}
```
Comment (fatih) [inline main.go:469]: I'm not a huge fan of goto's. Can you please change this so it doesn't use this pattern? You can create a small function that would return `t`

Comment (LeGEC) [reply main.go]: (I completely missed your notification) updated

**Comment by fatih** [comment]
Thank you @LeGEC for the contribution. I pushed another commit that improves some naming and adds comments. I'm planning to merge it this week.

**Pull request merged**